### ISSUE TITLE
simtools-simulate-prod does not accept required parameters defined through config file

### ISSUE THREAD

**Issue by VictorBarbosaMartins** [gammasim/simtools#792]
Perhaps an issue, but perhaps this is intentional:

- [ ] **Issue 1**:
` --primary, --azimuth_angle, --zenith_angle` are required for running `simtools-simulate-prod`. However, these arguments may also be defined in the production config file instead (`--production_config prod_config_proton_mini.yml` as in my case). I think the code should recognize that these three required parameters were already passed in the config file.
- [ ] **Issue 2**:
running without passing --site leads to an error, even though site is also defined in the config file
- [x] **Issue 3**: (solved, see comment below)
already for nshow=10 it seems the output it empty (from the config file) and nshow=100 from the command line_

**Comment by orelgueta** [comment]
Yeah I would say it is an issue. I think @GernotMaier solved it for other applications by passing the config file through the configurator class, while in this case it bypasses it. Can you maybe look for an example Gernot implemented and implement the same here?

**Comment by VictorBarbosaMartins** [comment]
Also, running without passing `--site` leads to an error, even though site is also defined in the config file:

```
(env) simtools-simulate-prod --production_config prod_config_proton_mini.yml --model_version Prod5 --start_run 0 --run 1 --primary proton --azimuth 0 --zenith 20
Traceback (most recent call last):
  File "/workdir/env/bin/simtools-simulate-prod", line 8, in <module>
    sys.exit(main())
  File "/workdir/external/simtools/simtools/applications/simulate_prod.py", line 243, in main
    simulator = Simulator(
  File "/workdir/external/simtools/simtools/simulator.py", line 138, in __init__
    self._load_configuration_and_simulation_model(config_data, config_file)
  File "/workdir/external/simtools/simtools/simulator.py", line 189, in _load_configuration_and_simulation_model
    self._load_corsika_config_and_model(config_showers)
  File "/workdir/external/simtools/simtools/simulator.py", line 207, in _load_corsika_config_and_model
    self.site = names.validate_site_name(self._corsika_config_data.pop("site"))
  File "/workdir/external/simtools/simtools/utils/names.py", line 230, in validate_site_name
    return validate_name(name, all_site_names)
  File "/workdir/external/simtools/simtools/utils/names.py", line 274, in validate_name
    raise ValueError(msg)
ValueError: Invalid name None
```

**Comment by VictorBarbosaMartins** [comment]
> Yeah I would say it is an issue. I think @GernotMaier solved it for other applications by passing the config file through the configurator class, while in this case it bypasses it. Can you maybe look for an example Gernot implemented and implement the same here?

Indeed, but in the other cases there is no mix of config file and parameters (at least not ideally). if we pass something in the config file through the configurator, we do not check anymore whether the required input is there

**Comment by orelgueta** [comment]
I think the new implementation from Gernot does actually allow to mark parameters as required and if they are not given anywhere then the application crashes. It also has a hierarchy implemented such that command line arguments overwrite arguments in the config file. It really is the best solution and should be propagated here if possible.

If not, then dropping the requirement for the parameters would work, but I am not crazy about it because it seems like an interim solution instead of a final one.

**Comment by VictorBarbosaMartins** [comment]
Orel and I just found out the reason why my simulations with the `simulate_prod` tool were leading to empty files and his simulations weren't, although we used the same machinery. It turns out that my google drive application in mac, which I use to constantly backup my laptop, was uploading the new output file to the cloud as it was being written, which probably prevented simtelarray to continue writing on it. That can probably also happen if your laptop is connected to the backup disk while running a simulation. So issue 3 is not an issue
